dev/CommandBarFlyout/CommandBarFlyout.cpp

```diff
@@ -252,25 +248,129 @@ winrt::Control CommandBarFlyout::CreatePresenter()
         {
-            presenter2.IsDefaultShadowEnabled(false);
+            // During the SecondaryCommands's animations, drop shadows need to disappear.
+            // However, performing a Remove during Closing/Opening and a Add during Closed/Opened
+            // doesn't line up perfectly with the end of the animations. So, if there are animations
+            // we'll remove the shadow in the Closing and Opening events, but we'll put back the shadow
+            // inside of CommandBarFlyoutCommandBar's Storyboard Completed event handlers.
+            if (auto commandBar = winrt::get_self<CommandBarFlyoutCommandBar>(m_commandBar.get()))
+            {
+                if (commandBar->HasSecondaryOpenCloseAnimations())
+                {
+                    if (!AlwaysExpanded() && m_secondaryCommands.Size() > 0)
+                    {
+                        RemoveDropShadow();
+                    }
+                }
+            }
         }
     }
+    });
+
+    m_commandBarOpenedRevoker = commandBar->Opened(winrt::auto_revoke, {
+    [this](auto const&, auto const&)
+    {
+        if (winrt::IFlyoutBase5 thisAsFlyoutBase5 = *this)
+        {
+            // If we open the CommandBar, then we should no longer be in a transient show mode -
+            // we now know that the user wants to interact with us.
+            thisAsFlyoutBase5.ShowMode(winrt::FlyoutShowMode::Standard);
+        }
+
+        if (SharedHelpers::Is21H1OrHigher() && m_presenter)
+        {
+            if (winrt::IControl7 presenterControl7 = m_presenter.get())
+            {
+                if (auto commandBar = winrt::get_self<CommandBarFlyoutCommandBar>(m_commandBar.get()))
+                {
+                    if (m_secondaryCommands.Size() > 0 || commandBar->IsOpen())
+                    {
+                        if (commandBar->IsExpandedUp())
+                        {
+                            presenterControl7.CornerRadius({ 0,0,4,4 });
+                        }
+                        else
+                        {
+                            presenterControl7.CornerRadius({ 4,4,0,0 });
+                        }
+                    }
+                }
+            }
+        }
+    }
+    });
+
+    m_commandBarClosingRevoker = commandBar->Closing(winrt::auto_revoke, {
+    [this](auto const&, auto const&)
+    {
+        if (SharedHelpers::Is21H1OrHigher())
+        {
+            if (auto commandBar = winrt::get_self<CommandBarFlyoutCommandBar>(m_commandBar.get()))
+            {
+                if (commandBar->HasSecondaryOpenCloseAnimations())
+                {
+                    RemoveDropShadow();
+                }
+            }
+        }
+    }
+    });
+
+    m_commandBarClosedRevoker = commandBar->Closed(winrt::auto_revoke, {
+    [this](auto const&, auto const&)
+    {
+        // When >21H1, if the CommandBar is closed, set the presenter's corner radius to 4.
+        if (SharedHelpers::Is21H1OrHigher())
```

Comment:
Closed(), Opened(), Closing(), Opening() - all these handlers call `SharedHelpers::Is21H1OrHigher()`- would it be better to call this function once and then add handlers if it returns true?:)